### PR TITLE
ref(ui): Replace BetaTag w/ FeatureBadge

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This component can be used to indicate new / beta / alpha features

### CODE REVIEW

File: src/sentry/static/sentry/app/components/featureBadge.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  new: t('new'),
+};
+
+const FeaturedBadge = ({type, variant = 'badge', title, ...p}: Props) => {
+  const tooltipTitle = title === undefined ? defaultTitles[type] : title;
+
+  return (
+    <div {...p}>
+      <Tooltip title={tooltipTitle} disabled={!tooltipTitle} position="right">
+        <React.Fragment>
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/featureBadge.tsx:37]: The reason we have the fragment here is due to the child types on `Tooltip`. it doesn't like that the two nodes can resolve to `false`.

File: src/sentry/static/sentry/app/components/sidebar/sidebarItem.tsx [hunk 2]
```diff
@@ -114,19 +112,13 @@ const SidebarItem = ({
...
-                    {t('Beta')}
-                  </StyledTag>
-                )}
+                {isNew && <FeatureBadge type="new" />}
+                {isBeta && <FeatureBadge type="beta" />}
               </LabelHook>
             </SidebarItemLabel>
           )}
+          {collapsed && isNew && <CollapsedFeatureBadge type="new" />}
+          {collapsed && isBeta && <CollapsedFeatureBadge type="beta" />}
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/sidebar/sidebarItem.tsx:121]: Added indicator dot support

File: src/sentry/static/sentry/app/views/settings/components/settingsLayout.tsx [hunk 3]
```diff
@@ -66,6 +66,7 @@ const SidebarWrapper = styled('div')`
   background: ${p => p.theme.white};
   border-right: 1px solid ${p => p.theme.borderLight};
   padding: ${space(4)};
+  padding-right: ${space(2)};
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/settings/components/settingsLayout.tsx:69]: Give's more room for the slightly bigger new badge

File: src/sentry/static/sentry/app/views/settings/components/settingsNavItem.tsx [hunk 4]
```diff
@@ -22,13 +22,7 @@ const SettingsNavItem = ({badge, label, index, id, ...props}: Props) => {
...
 
   const renderedBadge =
-    badge === 'new' ? (
-      <StyledTag priority="warning" size="small" border>
-        {badge}
-      </StyledTag>
-    ) : (
-      <Badge text={badge} />
-    );
+    badge === 'new' ? <FeatureBadge type="new" /> : <Badge text={badge} />;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/settings/components/settingsNavItem.tsx:25]: This is nicer since we kill the `priority="warning"`

File: src/sentry/static/sentry/app/components/featureBadge.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+import theme from 'app/utils/theme';
+import CircleIndicator from 'app/components/circleIndicator';
+
+type BadgeProps = {
+  type: 'alpha' | 'beta' | 'new';
+  variant?: 'indicator' | 'badge';
+  title?: string | null;
+};
+
+type Props = Omit<React.HTMLAttributes<HTMLDivElement>, keyof BadgeProps> & BadgeProps;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/featureBadge.tsx:17]: never used this before.. nice!! 👏

File: src/sentry/static/sentry/app/components/featureBadge.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+};
+
+const labels = {
+  alpha: t('alpha'),
+  beta: t('beta'),
+  new: t('new'),
+};
+
+const FeaturedBadge = ({type, variant = 'badge', title, ...p}: Props) => {
+  const tooltipTitle = title === undefined ? defaultTitles[type] : title;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/featureBadge.tsx:32]: shouldn't you also check if the title is `null`? 
```suggestion
  const tooltipTitle = !defined(title) ? defaultTitles[type] : title;
```

File: src/sentry/static/sentry/app/components/featureBadge.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+            <CircleIndicator color={theme.badge[type].indicatorColor} size={8} />
+          )}
+        </React.Fragment>
+      </Tooltip>
+    </div>
+  );
+};
+
+const StyledTag = styled(Tag)`
+  padding: 3px ${space(0.75)};
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/featureBadge.tsx:49]: has it to be really 3px? maybe you could use space(0.25) or space(0.5)?

File: src/sentry/static/sentry/app/components/featureBadge.tsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+import Tooltip from 'app/components/tooltip';
+import space from 'app/styles/space';
+import {t} from 'app/locale';
+import theme from 'app/utils/theme';
+import CircleIndicator from 'app/components/circleIndicator';
+
+type BadgeProps = {
+  type: 'alpha' | 'beta' | 'new';
+  variant?: 'indicator' | 'badge';
+  title?: string | null;
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/featureBadge.tsx:14]: Please why do you prefer to use null? I would only make the prop optional here

File: src/sentry/static/sentry/app/utils/theme.tsx [hunk 9]
```diff
@@ -112,10 +112,25 @@ const alert = {
...
   beta: {
     background: `linear-gradient(90deg, ${colors.pink}, ${colors.purple})`,
+    indicatorColor: colors.purple,
+    border: null,
   },
-} as const;
+  new: {
+    background: `linear-gradient(90deg, ${colors.green}, ${colors.greenDark})`,
+    indicatorColor: colors.green,
+    border: null,
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/utils/theme.tsx:131]: do you really need null here?

File: src/sentry/static/sentry/app/views/settings/components/tag.tsx [hunk 10]
```diff
@@ -1,22 +1,27 @@
...
 import InlineSvg from 'app/components/inlineSvg';
+import {Theme} from 'app/utils/theme';
 
-const getMarginLeft = p => {
-  if (!p.inline) {
-    return '';
-  }
-  return `margin-left: ${p.size === 'small' ? '0.25em' : '0.5em'};`;
+type Props = React.HTMLAttributes<HTMLDivElement> & {
+  priority?: keyof Theme['badge'] | keyof Theme['alert'];
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/settings/components/tag.tsx:8]: I would add an export in the theme like : 

```
export type Bagde = keyof typeof badges;
export type Alert = keyof typeof alerts;
```

File: src/sentry/static/sentry/app/views/settings/components/tag.tsx [hunk 11]
```diff
@@ -37,29 +42,20 @@ const Tag = styled(
   display: inline-flex;
   box-sizing: border-box;
   padding: ${p => (p.size === 'small' ? '0.1em 0.4em 0.2em' : '0.35em 0.8em 0.4em')};
-  font-size: 75%;
+  font-size: ${p => p.theme.fontSizeExtraSmall};
   line-height: 1;
-  color: ${p => (p.priority ? '#fff' : p.gray5)};
+  color: ${p => (p.priority ? '#fff' : p.theme.gray5)};
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/settings/components/tag.tsx:47]:
```suggestion
  color: ${p => (p.priority ? p.theme.white : p.theme.gray5)};
```

**Review by priscilawebdev — COMMENTED** [review]
Great Job 🚀 I just left a few suggestions

File: src/sentry/static/sentry/app/components/featureBadge.tsx (hunk 6 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/featureBadge.tsx:32]: No I actually want the null case to fall through to turn off the tooltip and not give a default tooltip. Maybe I can add a comment.

File: src/sentry/static/sentry/app/components/featureBadge.tsx (hunk 7 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/featureBadge.tsx:49]: haha yeah I know. It really just looks best at 3px and IMO at this small of values we have to prioritize visuals over consistent spacing units.

File: src/sentry/static/sentry/app/components/featureBadge.tsx (hunk 8 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/featureBadge.tsx:14]: `null` allows you to completely turn off the tooltip. I'll add a comment.

File: src/sentry/static/sentry/app/views/settings/components/tag.tsx (hunk 10 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/settings/components/tag.tsx:8]: 👎 would prefer not to pollute our namespace with additional types. Especially since we already have a component called alert.

What's the problem with doing this?

File: src/sentry/static/sentry/app/components/featureBadge.tsx (hunk 8 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/featureBadge.tsx:14]: What happens with empty string? Would a specific prop be better instead of overloading `title`?

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/featureBadge.tsx:14]: Yeah. I was just keeping bc since we did this one or two places. I'm good for a separate prop. fwiw I think empty string would also turn it off.

File: src/sentry/static/sentry/app/views/settings/components/tag.tsx [hunk 12]
```diff
@@ -1,22 +1,27 @@
...
+  inline?: boolean;
 };
 
-const getBorder = p =>
-  p.border
-    ? `border: 1px solid ${
-        p.priority ? p.theme.alert[p.priority].border : p.theme.gray1
-      };`
-    : '';
+type StyleFucProps = Props & {theme: Theme};
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/components/tag.tsx:15]: What's a StyleFucProps?

File: src/sentry/static/sentry/app/views/settings/components/tag.tsx [hunk 13]
```diff
@@ -1,22 +1,27 @@
...
+
+const getMarginLeft = (p: StyleFucProps) =>
+  p.inline ? `margin-left: ${p.size === 'small' ? '0.25em' : '0.5em'};` : '';
+
+const getPriority = (p: StyleFucProps) =>
+  p.priority ? p.theme.alert[p.priority] ?? p.theme.badge[p.priority] ?? null : null;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/components/tag.tsx:21]: this is kind of hard to parse

File: src/sentry/static/sentry/app/views/settings/components/tag.tsx (hunk 12 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/settings/components/tag.tsx:15]: missing n, embarrassing

File: src/sentry/static/sentry/app/views/settings/components/tag.tsx (hunk 13 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/settings/components/tag.tsx]: Yeah I made it a bit more obvious and added a comment

File: src/sentry/static/sentry/app/views/settings/components/tag.tsx (hunk 11 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/settings/components/tag.tsx:55]: I'll get this in another PR unless there's anything else worth changing in this one, would like to merge now while the tests pass (assuming I can get review)

**Review by billyvg — APPROVED** [review]
